### DISCUSSION TITLE
Gemini Error: Please ensure that multiturn requests ends with a user role or a function response.

### DISCUSSION THREAD

**Discussion by Bobeta** [danny-avila/LibreChat#1561] [Troubleshooting]
### What happened?

Using Naga AI and Gemini pro, when I write answers where the API needs to consider multiple messages, I get this error:
```
LibreChat         | 2024-01-15 09:54:26 warn: [OpenAIClient.chatCompletion][stream] API error
LibreChat         | 2024-01-15 09:54:26 warn: [OpenAIClient.chatCompletion][finalChatCompletion] API error
LibreChat         | 2024-01-15 09:54:26 error: [OpenAIClient.chatCompletion] Unhandled error type Error: 500 We got a bad response from the source. Status 400. Error: {
LibreChat         |   "error": {
LibreChat         |     "code": 400,
LibreChat         |     "message": "Please ensure that multiturn requests ends with a user role or a function response.",
LibreChat         |     "status": "INVALID_ARGUMENT"
LibreChat         |   }
LibreChat         | }
LibreChat         |
LibreChat         | 2024-01-15 09:54:26 error: [handleAbortError] AI response error; aborting request: 
```

After that, no matter what message I type in the chat, I get the same error using gemini pro, changing language model it works fine again. Opening a new chat with gemini pro makes it work again.

### Steps to Reproduce

Step 1) Use Naga AI and gemini pro
Step 2) Write answers where the API needs to consider multiple returns 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I think this is an issue when using Gemini under "OpenAI" or custom endpoint handling. 

This issue is not specific to LibreChat: https://github.com/songquanpeng/one-api/issues/836 https://github.com/babaohuang/GeminiProChat/issues/28

When using the "Google" endpoint this is handled for you. The error "Please ensure that multiturn requests ends with a user role or a function response." comes from Google API and I will investigate how to handle it when using a reverse proxy.
